am-lib/src/test/java/uk/gov/hmcts/reform/amlib/PermissionTest.java

```diff
@@ -0,0 +1,122 @@
+package uk.gov.hmcts.reform.amlib;
+
+import org.junit.Test;
+import uk.gov.hmcts.reform.amlib.enums.Permission;
+import uk.gov.hmcts.reform.amlib.exceptions.UnsupportedPermissionsException;
+import uk.gov.hmcts.reform.amlib.utils.Permissions;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+@SuppressWarnings("PMD")
+public class PermissionTest {
+
+    @Test
+    public void sumOf_whenPassingPermissions_theSumOfValuesIsCalculated() {
+        Set<Permission> permissions = Stream.of(Permission.CREATE, Permission.READ).collect(Collectors.toSet());
+
+        int sum = Permissions.sumOf(permissions);
+        int expectedSum = Permission.CREATE.getValue() + Permission.READ.getValue();
+
+        assertThat(sum).isEqualTo(expectedSum);
+    }
+
+    @Test
+    public void sumOf_whenPassingNoPermissions_theSumOfValuesIsZero() {
+        Set<Permission> permissions = new HashSet<>();
+
+        int sum = Permissions.sumOf(permissions);
+
+        assertThat(sum).isEqualTo(0);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueOne_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(1);
+
+        assertThat(permissions).containsOnly(Permission.CREATE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueTwo_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(2);
+
+        assertThat(permissions).containsOnly(Permission.READ);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueThree_ExpectArray() throws UnsupportedPermissionsException {
```

Comment:
Please add tests for sum equal 6, 8, 10, 11, 13.

Reply:
👍